docs/gen_ref_pages.py

```diff
@@ -0,0 +1,79 @@
+"""
+Generate the code reference pages.
+Based off https://mkdocstrings.github.io/recipes/#automatic-code-reference-pages
+"""
+
+from pathlib import Path
+
+import mkdocs_gen_files
+
+# need to adjust to the test suite hook
```

Comment:
There seem to be quite some comments that are no longer relevant in this `gen_ref_pages.py`. Would be good to clean those up.